### ISSUE TITLE
Switch node-promise with bluebird

### ISSUE THREAD

**Issue by marcofranssen** [yfinkelstein/node-zookeeper#118]
As bluebird is the promise library which has the best results in benchmarks and is widely know, I suggest to replace node-promise with this one. It also implements the A+ spec for promises meaning you can easily replace it with es6 promises as they also implement the A+ spec.

Bluebird supports many other node libraries, so you can easily promisify them.

Here some examples on how to use it with other 
http://bluebirdjs.com/docs/features.html

Basically make sure this module can deal with bluebird.promisify or promisifyAll and we are good to go. Then ask the bluebird guys to also list node-zookeper in their docs ;-)

**Comment by DavidVujic** [comment]
We recently dropped support for node versions prior to 8 and that will give us an opportunity to rewrite the code using ES 201x syntax. Promises are a native feature and I think we should give that a try. I'm closing this issue, thank you for contributing with ideas and suggestions. I am a new maintainer and would be very happy for more feedback and contributions!
